Lib/test/test_socket.py

```diff
@@ -52,6 +52,28 @@
 except ImportError:
     _socket = None
 
+_hunting_for_refleaks = None
+def hunting_for_refleaks():
+    """
+    Return true iff running tests while hunting for refleaks
+    """
+    from test.libregrtest.runtests import RunTests
+    import gc
+
+    global _hunting_for_refleaks
+
+    if _hunting_for_refleaks is None:
+        for value in gc.get_objects():
+            if isinstance(value, RunTests):
+                _hunting_for_refleaks = (value.hunt_refleak is not None)
+                break
+        else:
+            _hunting_for_refleaks = False
+
+    return _hunting_for_refleaks
+
+
+
```

Comment:
This function is an awful hack and should not be merged in its current form.

If we want to have a support function for detecting a refleak hunting run it should be part of `test.support` and have a clean interface to the test runner.